### PR TITLE
TSK-989 Filter Tasks by planned TimeIntervals (REST)

### PR DESCRIPTION
**Pull request by gitgoodjhe**
-Added planned param in filter list
-Implemented needed methods
-Added Test

### CODE REVIEW

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/TaskController.java [hunk 1]
```diff
@@ -330,13 +332,65 @@ private TaskQuery applyFilterParams(TaskQuery taskQuery, MultiValueMap<String, S
...
+    TimeInterval[] timeIntervalArray = new TimeInterval[timeIntervalsList.size()];
+
+    TimeInterval[] timeIntervals = timeIntervalsList.toArray(timeIntervalArray);
+
+    return timeIntervals;
+  }
+
+  private TimeInterval determineTimeInterval(String[] instants, int i) {
+
+    if (!instants[i].equals("maxTime") && !instants[i + 1].equals("maxTime")) {
```
Comment (benjamineckstein) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/rest/TaskController.java]: please consider making maxTime some kind of constant. also is this somewhere documented within the API documentation? this is like a magic string that suddenly does not need to be a valid timestamp / instant object and still have a meaning. 
Personally i would just remove this special cases and find a way to give "null" or only 1 timestamp such a meaning.

File: rest/taskana-rest-spring/src/test/resources/asciidoc/rest-api.adoc [hunk 2]
```diff
@@ -115,6 +115,26 @@ include::{snippets}/GetAllTasksDocTest/response-fields.adoc[]
...
+planned | plannedFrom | plannedTo | due | dueFrom | dueTo +
+
+If it is sufficient to filter the list with a single time interval, use the parameters +
+plannedFrom/dueFrom and plannedTo/plannedTo. +
+Leave one of them out of the request, if you want to have an open interval. +
+
+If it is required to filter the list with multiple planned or due time intervals, the parameters +
+ planned / due can be used as follows instead: +
+
+ - provide a string with an even number of instants separated by a ",". Time intervals will be +
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/resources/asciidoc/rest-api.adoc]: provide this example with an example request? Could be better to understand.

File: rest/taskana-rest-spring/src/test/resources/asciidoc/rest-api.adoc [hunk 3]
```diff
@@ -115,6 +115,26 @@ include::{snippets}/GetAllTasksDocTest/response-fields.adoc[]
 The list generated in the response can be filtered using following parameters in the uri: +
 name | priority | state | classification.key | workbasket-id | {workbasket-key , domain} | +
 owner | por.company | por.system | por.instance | por.type | por.value +
+planned | plannedFrom | plannedTo | due | dueFrom | dueTo +
+
+If it is sufficient to filter the list with a single time interval, use the parameters +
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/resources/asciidoc/rest-api.adoc]: I know what you are trying to say. But it's hard to read. Maybe something like

> planned stands for a specific timestamp. Whenever an interval is of interest use plannedFrom and plannedTo. plannedFrom and plannedTo have to be used together. due works analog.

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/TaskController.java (hunk 1 above)
Comment (gitgoodjhe) [reply rest/taskana-rest-spring/src/main/java/pro/taskana/rest/TaskController.java]: Done. Also, we now offer 2 methods to filter. If one time interval is sufficient, the parameters "plannedFrom/dueFrom" and "plannedTo/dueTo" can be used. If one of those is left out it is an open time interval (like you suggested). For the other option to use multiple time intervals as filter, the parameter "planned" can be used and given an even amount of instants. For open time intervals an "empty" string can be used.

Comment (benjamineckstein) [reply rest/taskana-rest-spring/src/main/java/pro/taskana/rest/TaskController.java]: what happens if someone uses both combinations? -> plannedFrom and planned. Or plannedFrom, plannedTo and planned? 

What is the expected & current behaviour? We open a lot of edge cases that need to be defined and handled.

Comment (gitgoodjhe) [reply rest/taskana-rest-spring/src/main/java/pro/taskana/rest/TaskController.java]: There is a check for invalid parameter combos in the new commit.

**Pull request merged**